ext/byebug/byebug.c

```diff
@@ -579,7 +579,6 @@ Stop(VALUE self)
 
     breakpoints = Qnil;
     catchpoints = Qnil;
-    threads = Qnil;
```

Comment:
`cleanup_dead_threads()` depends on `threads` availability. Because `release_lock()` calls `cleanup_dead_threads()`, immediately clearing `threads` causes nil reference by locked threads in `acquire_lock()`.

By keeping `threads`, we can safely release thread locks in `EVENT_TEARDOWN`. This variable is re-assigned with `create_threads_table()` in Byebug.start. So there is no need to clear `threads` in this place.